### DISCUSSION TITLE
Golang Support

### DISCUSSION THREAD

**Discussion by Faelivrinx** [LazyVim/LazyVim#503] [Q&A]
I'm looking forward any guide for configuring this with golang language. Would you recommend something?

**Comment by ueaner** [comment]
You can refer to [here](https://github.com/ueaner/nvimrc/blob/main/lua/plugins/extras/lang/example.lua). You know that LazyVim already supports [LSP](https://www.lazyvim.org/plugins/lsp) and [TreeSitter](https://www.lazyvim.org/plugins/treesitter), you only need to care about [dap.lua](https://github.com/ueaner/nvimrc/blob/main/lua/plugins/dap.lua) and [test.lua](https://github.com/ueaner/nvimrc/blob/main/lua/plugins/test.lua) and [example.lua](https://github.com/ueaner/nvimrc/blob/main/lua/plugins/extras/lang/example.lua) (for the golang configuration example). I hope it can help you.

**Comment by amaanq** [comment]
LSP: add gopls
Plugins: go.nvim, and nvim-dap-go are good contenders and you should be set

**Reply by robsonpeixoto** [reply]
Could you share how did you configured the `ray-x/go.nvim` ?

**Reply by binary4cat** [reply]
I use the default configuration, there is no problem, some personalized ones can be customized by referring to the plugin documentation.

**Reply by amaanq** [reply]
yeah same, I just set config = true, though I removed it now as I didn't feel like I needed it

**Comment by fredrikaverpil** [comment]
Did any of you enable the LazyVim Go extras?
[It doesn't seem like](https://github.com/LazyVim/LazyVim/discussions/1107) the DAP is set up there. How would you have set it up?

**Reply by binary4cat** [reply]
I enabled it, and also enabled dap extras, nothing is configured, and the go program can be debugged directly.

**Reply by fredrikaverpil** [reply]
> I enabled it, and also enabled dap extras, nothing is configured, and the go program can be debugged directly.

How did you invoke the debugger?

I opened a PR so to enable `<leader>td` which was how I expected for it to work: https://github.com/LazyVim/LazyVim/pull/1115

**Reply by binary4cat** [reply]
The cursor is on the unit test that needs to be debugged, and use the shortcut key `<leader>dc` to start the debugger, and then select `Delve: Debug test (go.mod)`, and I can start debugging normally.

**Reply by folke** [reply]
@binary4cat that is without the new PR?

**Reply by dpetka2001** [reply]
This feature must be coming from `delve` itself and its nvim-dap configuration. The PR is to address the lack of support from neotest go adapter about dap configuration, thus the lack of calling neotest-go with `strategy=dap`. It seems to me that you can achieve the same thing in different ways. Either use default `nvim-dap` configuration to debug test or if you want to use neotest no matter what then you should use the way that PR is doing it. I don't know if there are other major differences between the 2 ways of achieving the same thing.

**Reply by fredrikaverpil** [reply]
I've gotten used to debugging a test with `<leader>td` in multiple languages and so I reacted to this not working with the LazyVim Go extras.

I just tried debugging a go test with `<leader>dc` and that works too (without the changes introduced in #1115). But I would personally still want it to work on `<leader>td`.

**Reply by binary4cat** [reply]
> I just tried debugging a go test with `<leader>dc` and that works too (without the changes introduced in https://github.com/LazyVim/LazyVim/pull/1115).

Yes, as @fredrikaverpil  said, can directly debug go test with `<leader>dc`.

**Comment by Nikola-Milovic** [comment]
Since the [`extras.lang.go`](https://www.lazyvim.org/extras/lang/go) has basically everything, how can one enable autoimporting via `goimports` on save? I used to use `ray-x/go.nvim` for my go setup but with the extras I already have everything that I'd need practically.

**Reply by fredrikaverpil** [reply]
@Nikola-Milovic I have this on save. I've installed it with Mason and I use `none-ls` (the fork of null-ls) for it:

```lua
require("null-ls").builtins.formatting.goimports
```
